src/services/events/events-schema.ts

```diff
@@ -0,0 +1,36 @@
+import mongoose from "mongoose";
+import { Schema } from "mongoose";
+import { z } from "zod";
+
+export const EventInfo = z.object({
+    eventId: z.coerce.string().cuid2(),
+    name: z.string(),
+    date: z.string().transform((str) => new Date(str)),
+    start_time: z.string().transform((str) => new Date(str)),
+    end_time: z.string().transform((str) => new Date(str)),
+});
+
+export const EventSchema = new Schema({
+    eventId: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    name: {
+        type: String,
+        required: true,
+    },
+    date: {
+        type: Date,
+        required: true,
+    },
+    start_time: {
+        type: Date,
+        required: true,
+    },
+    end_time: {
+        type: Date,
+        required: true,
+    },
+});
+export const Event = mongoose.model("Event", EventSchema);
```

Comment:
we should not create a model here. instead, it should be imported in the predefined database file and saved as a model. (this is going to make our lives a lot easier when we do start testing infra) [initializeModel](https://github.com/ReflectionsProjections/rp-api/blob/618f2edd5e52a774b8a06c1efaafa2c10a400021/src/database.ts#L6C10-L6C25) function